connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -946,7 +946,7 @@ public Void call() throws Exception {
                 } else if (isLeader()) {
                     callback.onCompletion(new NotAssignedException("Cannot restart task since it is not assigned to this member", member.ownerUrl(id)), null);
                 } else {
-                    callback.onCompletion(new NotLeaderException("Cannot restart task since it is not assigned to this member", leaderUrl()), null);
+                    callback.onCompletion(new NotLeaderException("Cannot restart task since it is not leader", leaderUrl()), null);
```

Comment:
I think this change is actually less clear. Workers (members of the cluster) can be leaders, while connectors and tasks are assigned to workers. But connectors/tasks are not workers (members), so it's actually incorrect to say "Cannot restart task since it [the task] is not a leader".

About the only improvement to the original message that adds value is to change "member" to "worker", though the value is debatable since we do sometimes use "member" in other log messages (including in this class!).

Reply:
I dropped the 2nd and 3rd hunks since it is not clear how to make the messages more helpful.
Please take another look.